great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -423,152 +480,187 @@ def _reconcile_rule(
-                batch_request=parameter_builder.batch_request
+        current_parameter_builders: Dict[
+            str, ParameterBuilder
+        ] = rule._get_parameter_builders_as_dict()
+
+        parameter_builder_name: str
+        parameter_builder: ParameterBuilder
+        parameter_builder_as_dict: dict
+        for (
+            parameter_builder_name,
+            parameter_builder,
+        ) in current_parameter_builders.items():
+            parameter_builder_as_dict = parameter_builder.to_dict()
+            parameter_builder_as_dict[
+                "class_name"
+            ] = parameter_builder.__class__.__name__
+            parameter_builder_as_dict[
+                "module_name"
+            ] = parameter_builder.__class__.__module__
+
+            # Roundtrip through schema validation to add any missing fields
+            deserialized_config: dict = parameterBuilderConfigSchema.load(
+                parameter_builder_as_dict
+            )
+            serialized_config: dict = parameterBuilderConfigSchema.dump(
+                deserialized_config
             )
 
-        effective_parameter_builder_config["batch_request"] = batch_request
+            effective_parameter_builder_configs[
+                parameter_builder_name
+            ] = serialized_config
 
-        key: str
-        value: Any
-        current_value: Any
-        for key, value in parameter_builder_config.items():
-            if hasattr(parameter_builder, f"{key}"):
-                current_value = getattr(parameter_builder, key)
-                effective_parameter_builder_config[key] = value or current_value
-            else:
-                effective_parameter_builder_config[key] = value
+        effective_parameter_builder_configs = nested_update(
+            effective_parameter_builder_configs,
+            {
+                parameter_builder_config["name"]: parameter_builder_config
+                for parameter_builder_config in parameter_builder_configs
+            },
+            dedup=True,
+        )
 
-        return effective_parameter_builder_config
+        if not effective_parameter_builder_configs:
+            return None
 
-    @staticmethod
-    def _reconcile_expectation_configuration_builder_configs_for_rule(
-        rule: Rule, rule_config: dict
-    ) -> List[dict]:
-        effective_expectation_configuration_builder_configs: List[dict] = []
-        expectation_configuration_builder_configs: List[dict] = rule_config.get(
-            "expectation_configuration_builders", []
-        )
+        return list(effective_parameter_builder_configs.values())
 
-        current_expectation_configuration_builders: Dict[
-            str, ExpectationConfigurationBuilder
-        ] = rule.expectation_configuration_builders
+    @staticmethod
+    def _reconcile_rule_expectation_configuration_builder_configs(
+        rule: Rule, expectation_configuration_builder_configs: List[dict]
+    ) -> Optional[List[dict]]:
```

Comment:
Instead of an `Optional` list, can we just return an empty list?

Reply:
@cdkini Good idea -- done (in next commit) -- thanks!